cargo-cyclonedx/src/generator.rs

```diff
@@ -311,6 +316,15 @@ impl SbomGenerator {
 
         metadata.tools = Some(Tools(vec![tool]));
 
+        let properties = match self.config.target.as_ref().unwrap() {
+            SingleTarget(target) => vec![Property::new("rustcTarget", &target)],
+            AllTargets => all_known_targets()
+                .into_iter()
+                .map(|target| Property::new("rustcTarget", &target))
```

Comment:
I like the idea in general. Have you checked if there are any plans/standards/best practices for what this property could be called? Other tools must have the same idea/concept in general (e.g. Go/C/C++)

Reply:
I have not really checked for other uses of `properties`, just skimmed the standardized fields of `metadata`. 

This encodes Rust target triples specifically, e.g. `x86_64-unknown-linux-gnu`, so this is not going to be interoperable with anything else. Hence the use of `properties` instead.

Reply:
Understood. In that case, does it make sense to use a namespaced property instead?

https://github.com/CycloneDX/cyclonedx-property-taxonomy
https://github.com/CycloneDX/cyclonedx-property-taxonomy/blob/main/cdx.md

Reply:
Found the property for Go: https://github.com/CycloneDX/cyclonedx-property-taxonomy/blob/main/cdx/gomod.md

Reply:
Yes, it does. Good point.